relay-event-schema/src/protocol/security_report.rs

```diff
@@ -1080,6 +1144,12 @@ impl SecurityReportType {
 
         Ok(if helper.csp_report.is_some() {
             Some(SecurityReportType::Csp)
+        } else if let Some(ty) = helper.ty {
+            if *ty == "csp-violation" {
```

Comment:
Nit, we could actually make the type an enum?

Reply:
I think the type will be always the same. But I'm also fine with creating an enum for this. And then also remove `Cow<...>` altogether.